invenio_drafts_resources/services/records/components/files.py

```diff
@@ -64,6 +67,14 @@ def update_draft(self, identity, data=None, record=None, errors=None):
                     ],
                 }
             )
+        elif draft.files.enabled and draft.files.items():
+            for item in draft.files.items():
+                data = item[1].file.dumps()
```

Comment:
question: why accesing `[1]`? I'm not a 100% of the types. But I think we could iterate over [`draft.files` (without `.items()`)](https://github.com/inveniosoftware/invenio-rdm-records/blob/ab4390dbcf264b3dc63140e0ad6f1e541747f4b7/invenio_rdm_records/records/api.py#L244). This would return [`FileRecord` objects](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/records/api.py#L51), from which we can access the actual [`file` instance](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/records/api.py#L71). Then due to the overwrite of [`__getattr__` ](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/records/api.py#L171) we can access the uri.

```python
for file_record in draft.files:
   uri = file_record.file.ui
   if not uri.is_completed:
       ...
```

Important: Im describing based on memory and class type assumptions I can be fully wrong.

Reasoning behind all this:
- In general I would try to avoid hardcoding list indixes (i.e. `[1]`) beause they can change and then we won't know we referenced it here and things will break.
- As much as possible avoid using `dumps` and access properties directly. Due to how descriptors work, the properties should be available (if not we might need to check that).